webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js

```diff
@@ -36,6 +36,19 @@ const buildTableRow = (subscription, availableQuantities, updatedQuantity) => {
 const buildTableRowsFromGroup = (subscriptionGroup, availableQuantities, updatedQuantity) => {
   const { open, subscriptions } = subscriptionGroup;
 
+  if (subscriptionGroup.subscriptions.length > 1) {
+    const temp = JSON.parse(JSON.stringify(subscriptionGroup.subscriptions[0]));
+    temp.id=0
+    temp.contract_number = 'NA';
+    temp.start_date = 'NA';
+    temp.end_date = 'NA';
+    temp.consumed = 'NA';
+
+    if (subscriptionGroup.subscriptions[0].id != 0) {
+      subscriptionGroup.subscriptions.unshift(temp);
```

Comment:
I don't really like how we're creating a 'fake' subscription by adding it to the subscriptions list to create a row - but it does work!

Here's another approach to get around modifying the subscriptions array which I think is better practice to keep our data and view separate:

```
diff --git a/webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js b/webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js
index f4f1a3b..f580299 100644
--- a/webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js
+++ b/webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js
@@ -33,13 +33,33 @@ const buildTableRow = (subscription, availableQuantities, updatedQuantity) => {
   return baseSubscription;
 };
 
+const buildTableCollapseRow = (subscriptionGroup) => {
+  const { subscriptions } = subscriptionGroup;
+  const first = subscriptions[0];
+
+  let heading = {};
+  heading.contract_number = 'NA';
+  heading.start_date = 'NA';
+  heading.end_date = 'NA';
+  heading.consumed = 'NA';
+  heading.product_id = first.product_id;
+  heading.name = first.name;
+
+  return heading;
+}
+
 const buildTableRowsFromGroup = (subscriptionGroup, availableQuantities, updatedQuantity) => {
   const { open, subscriptions } = subscriptionGroup;
 
-  // build row for each subscription
-  if (open) {
-    return subscriptions.map(subscription =>
-      buildTableRow(subscription, availableQuantities, updatedQuantity));
+  let rows = [];
+
+  if (subscriptions.length > 1) {
+    rows.push(buildTableCollapseRow(subscriptionGroup));
+
+    // build row for each subscription
+    if (open) {
+      return rows.concat(subscriptions.map(sub => buildTableRow(sub, availableQuantities, updatedQuantity)));
+    }
   }
 
   // build row only for the first subscription in the group
diff --git a/webpack/scenes/Subscriptions/components/SubscriptionsTable/components/Table.js b/webpack/scenes/Subscriptions/components/SubscriptionsTable/components/Table.js
index 33d6465..81e88a5 100644
--- a/webpack/scenes/Subscriptions/components/SubscriptionsTable/components/Table.js
+++ b/webpack/scenes/Subscriptions/components/SubscriptionsTable/components/Table.js
@@ -29,7 +29,7 @@ const Table = ({
   const groupingController = {
     isCollapseable: ({ rowData }) =>
       // it is the first subscription in the group
-      rowData.id === groupedSubscriptions[rowData.product_id].subscriptions[0].id &&
+      //rowData.id === groupedSubscriptions[rowData.product_id].subscriptions[0].id &&
       // the group contains more then one subscription
       groupedSubscriptions[rowData.product_id].subscriptions.length > 1,
     isCollapsed: ({ rowData }) => !groupedSubscriptions[rowData.product_id].open,
```
I don't think what I pasted is fully working. We'll also want to remove the checkbox from the collapsible row since it doesn't really have a purpose (it's not an actual subscription).

Let me know if I didn't explain this well.

Reply:
@jturel I completely agree with you. I tried doing something along the same lines and it was not working properly. So I switched to this solution. But I do agree that creating a fake subscription was not right.